src/test/java/ClientConnectionTest.java

```diff
@@ -0,0 +1,60 @@
+import echoserver.ClientConnection;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+class FakeSocket extends Socket{
```

Comment:
Can you talk about the decision to extend vs introducing an interface?

Reply:
+1